### PR TITLE
Upgrade otel to v0.29.0

### PR DESCRIPTION
**Pull request by mapno**
#### PR Description 

Upgrade OTel from `v0.21.0` to `v0.29.0`. It uses an internal fork that removes internal folders. A first attempt to upgrade was done in https://github.com/grafana/agent/pull/641, but finally discarded because the new interface we had to use wasn't configurable to our needs. That fork is a temporary solution (see [issue](https://github.com/open-telemetry/opentelemetry-collector/issues/3482) to improve configuration)

#### PR Checklist

- [x] CHANGELOG updated 
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by joe-elliott — APPROVED** [review]
nice!

File: go.mod [hunk 1]
```diff
@@ -109,4 +108,10 @@ exclude google.golang.org/grpc/examples v0.0.0-20200728065043-dfc0c05b2da9
 replace github.com/weaveworks/common => github.com/rfratto/weaveworks-common v0.0.0-20210326192855-c95210d58ba7
 
 // loadbalancingexporter uses non-fixed version of batchpertrace which fetches latest and causes problems
-replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpertrace => github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpertrace v0.21.0
+replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpersignal => github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpersignal v0.29.0
+
+replace go.opentelemetry.io/collector => github.com/grafana/opentelemetry-collector v0.4.1-0.20210625154936-5ed8906327ca
```
Comment (rfratto) [inline go.mod:113]: Can we add a comment to this explaining that it's temporary and the issue we're waiting on?

**Review by rfratto — APPROVED** [review]
LGTM! I'd like to see a comment explaining the replace but other than that, merge at will :)

File: go.mod (hunk 1 above)
Comment (mapno) [reply go.mod:116]: Fixed in 409252146891ffee7ae595ef666b948223679d3f.

**Pull request merged**